detect_meteors.py

```diff
@@ -176,11 +182,52 @@ def detect_meteors(rf_dir, id_dir, noise_dir, output_dir,
         tmm.loadFromHdf5('/tmp/tmm.hdf5', skip_lowlevel=True)
     else:
         tmm.loadFromHdf5(skip_lowlevel=True)
-
-    for k, (tx, rx) in enumerate(data_generator(rfo, ido, no, tmm, s0, s1, rxch, txch)):
-        #FIXME call processing functions here
-        pass
-
+    
+    data_list = []
+    saved_data = np.zeros((59858, 1120))
+    pulse_data = data_generator(rfo, ido, no, tmm, s0, s1, rxch, txch)
+    times = []
+    for k, (tx, rx) in enumerate(pulse_data):
+
+        num_of_pulses = np.int((s1 - s0)/(rx.sample_rate*0.0267300605774))
+        # function that calculates snr values
+        def data_cal(array):
+            snr_vals = (np.abs(array.values)**2)/rx.noise_power
+            snr_point = np.unravel_index(np.argmax(snr_vals), (6879, 480))
+            max_snr = snr_vals[snr_point[0], snr_point[1]]
+            return max_snr, snr_point, snr_vals
+
+        mf_rx = freq_dft(tx, rx)
+        max_snr, snr_point, snr_vals = data_cal(mf_rx)
+        meteor_list = is_there_a_meteor(mf_rx, max_snr, snr_point, 1, 20533, 211200)
+
+        def list_filter(data):
+	    velocity = mf_rx.frequency.values[snr_point[1]]*3e8/(440e6*2)
+            info = (velocity, max_snr, k)
+            data.extend(info)
+            return data
+
+        if meteor_list != []:
+            new_list = list_filter(meteor_list)
+            data_list.append(new_list) 
+
+        saved_data[k, :] = snr_vals[480 : 1600, snr_point[1]]
+        range_vals =(3e8*mf_rx.delay.values[480 : 1600])/(2*rx.sample_rate)
+        times.append(mf_rx.t.values)
+
+    clustering = Clustering(eps=12, min_samples=1, tscale=0.03, rscale=150, vscale=710.27)
+    def cluster_generator(data):
+    	for item in data:
+            for c in clustering.addnext(t=item[0], r=item[1], v=item[3], snr=item[4], pulse_num=item[5]):
+                yield c
+        for c in clustering.finish():
+            yield c
+
+    clusters = list(cluster_generator(data_list))
```

Comment:
So right now this is running through all of the data and getting a big list of all possible meteor detections in individual pulses. Then it is running through that big list and performing clustering. We expect to run this on large data sets where we might not have enough memory to do it this way. Instead, we need to cluster incrementally as we go, which is why we are using this particular clustering algorithm in the first place.

So, we need to define the clustering object before the loop that reads the data. Then as soon as we get a meteor_list for a single pulse from the detection function, we feed it to the clustering algorithm with addnext. This will return any new clusters that have been formed from previously-added data. Then you can take that list of one or more new clusters and summarize them. Once summarized, then you can add the result to a CSV file. That file will have to be initialized before the loop as well, since we'll want to write to it incrementally. (In the future, we may decide another type of file would be more appropriate.) Then once the loop has finished, we have to finish the clustering by calling the finish function, which may return some final new clusters to summarize and save.

Switching to this incremental approach may also help save processing time by reducing memory requirements.